### ISSUE TITLE
Control DXR possible memory leak

### ISSUE THREAD

**Issue by screwylightbulb** [HansKristian-Work/vkd3d-proton#571]
Super hard to replicate. I got an out of video memory error crash 2 times yesterday while using DXR. Happened today too, but I was playing over an hour before it happened. Nothing useful in the log, and looking at the graph, it's not like a massive spike or anything. 

Actually it may be related to shader compilation, because it crashed today as I was entering a different area. 

I am on:
Kernel 5.10 
Nvidia driver 460.56
RTX3060Ti

Proton 5.13-6

![image](https://user-images.githubusercontent.com/6892136/111026238-e9ee5600-83f1-11eb-83ff-78da472a0b5c.png)

**Comment by doitsujin** [comment]
what does "out of video memory error crash" mean exactly? Is that a message from the game itself or from the vkd3d log?

**Comment by screwylightbulb** [comment]
This is from the game itself. A crash notification pops up with a message that says it is out of video ram. The log did not have anything in it worth mentioning. I could try upping the log level.

**Comment by HansKristian-Work** [comment]
If it's shader compilation, I expect you'd get a crash when entering a specific area or something.

**Comment by screwylightbulb** [comment]
> If it's shader compilation, I expect you'd get a crash when entering a specific area or something where the game needs some specific RT shaders or something we haven't seen yet.

Yeah, today it crashed with this while entering a new area. The 2 times it crashed yesterday was a different area, and it did not crash in that area today at all.

**Comment by vityafx** [comment]
What is your resolution? I couldn't run 4K at all with ray tracing enabled with my 10gb videocard, but on windows with dlss from 1440p it ran just fine (understandable why).

**Comment by screwylightbulb** [comment]
> What is your resolution?

1440. I could try with 1080.

**Comment by vityafx** [comment]
> > What is your resolution?
> 
> 
> 
> 1440. I could try with 1080.
> 
> 

I have started the full game from the beginning and have finished it fully with ray tracing on Linux at 1080p, with no crashes whatsoever, so I guess is that you just run out of memory because you have a fairly low ram capacity on your video card for these games at more or less high resolution including all these effects.

**Comment by Blisto91** [comment]
@screwylightbulb Friendly ping and a hello 👋 
Do you still experience this issue?

**Comment by HansKristian-Work** [comment]
Assuming to be fixed, and no activity in a year.
